### PR TITLE
Add rust src to package set

### PR DESCRIPTION
**Pull request by antigravityla**
This shortcut makes it easier to construct a shell with coordinated rust sources
for consumption in rust analyzer

The good part of https://github.com/tenx-tech/cargo2nix/pull/115/files

### CODE REVIEW

**Comment by antigravityla** [comment]
Lost at least one build to Ci.. local succeeds.  Cautious

File: overlay/make-package-set/full.nix [hunk 1]
```diff
@@ -11,6 +11,7 @@
   packageFun,
   cargo,
   rustc,
+  rust-src,
```
Comment (trha) [inline overlay/make-package-set/full.nix:14]: Let's just get rid of these and take `rustChannel` for simplicity.

Comment (antigravityla) [reply overlay/make-package-set/full.nix:14]: Just to simplify arguments?

**Comment by antigravityla** [comment]
> Is this a good idea? Do you know if this would lead to each crate derivation pulling in rust-src when copied between Nix stores when it's not strictly speaking needed, depending on whether we're cross-compiling or not?

I'm leaning on lazy evaluation AFAIK, so only if someone is consuming the attribute, but please elaborate if I've misinterpreted the behavior in the case that the user never writes `rust-src` anywhere

**Comment by psionic-k** [comment]
@antigravityla  closing in favor of upcoming rust-analyzer packaging & proper shell support

@ebkalderon to address your earlier concern I'll take a look at the closures for bin targets for example.

Upcoming PR will encompass concerns from #151 and #152, #145, this PR, and a few outstanding issues that make it difficult to employ cargo2nix as a solution for independent nix packaging of rust projects

**Pull request closed without merging**